sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/SchemaUtil.java

```diff
@@ -337,4 +337,33 @@ public Row mapRow(ResultSet rs) throws Exception {
       return rowBuilder.build();
     }
   }
+
+  /**
+   * compares two fields. Does not compare nullability of field types
+   * @param a field 1
+   * @param b field 2
+   * @return TRUE if fields are equal. Otherwise FALSE
+   */
+  public static boolean compareSchemaField(Schema.Field a, Schema.Field b) {
+    if (!a.getName().toLowerCase().equals(b.getName().toLowerCase()))
+      return false;
+    return compareSchemaFieldType(a.getType(), b.getType());
+  }
+
+  /**
+   * compares two FieldType. Does not compare nullability
+   * @param a FieldType 1
+   * @param b FieldType 2
+   * @return TRUE if FieldType are equal. Otherwise FALSE
+   */
+  public static boolean compareSchemaFieldType(Schema.FieldType a, Schema.FieldType b) {
+    if (a.getTypeName().equals(b.getTypeName()))
```

Comment:
The Field class already contains several comparison methods. I think it is prudent to use those methods rather than inventing your own.

Furthermore, you should keep in mind that database types usually have length restrictions. So a field of type STRING can only be assigned to a field of type VARCHAR(200) only if the length of the string is less than 200 characters. Have a look at the implementation of logical types in `LogicalTypes` class where the `equals` methods take this into account.